app/src/main/java/com/fcaputo/parkingapp/mvp/view/HomeView.kt

```diff
@@ -0,0 +1,38 @@
+package com.fcaputo.parkingapp.mvp.view
+
+import android.app.Activity
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import com.fcaputo.parkingapp.activity.ReservationActivity
+import com.fcaputo.parkingapp.activity.SettingsActivity
+import com.fcaputo.parkingapp.databinding.ActivityHomeBinding
+import com.fcaputo.parkingapp.mvp.contract.HomeContract
+import com.fcaputo.parkingapp.mvp.view.base.ActivityView
+
+class HomeView(activity: Activity) : ActivityView(activity), HomeContract.View {
+    private var binding: ActivityHomeBinding = ActivityHomeBinding.inflate(activity.layoutInflater)
+
+    init {
+        activity.setContentView(binding.root)
+    }
+
+    override fun onSettingsButton(onClick: () -> Unit) {
+        binding.btnSettings.setOnClickListener { onClick() }
+    }
+
+    override fun onMakeReservationButton(onClick: () -> Unit) {
+        binding.btnReservation.setOnClickListener { onClick() }
+    }
+
+    override fun navigateToSettings() {
+        Intent(activity as AppCompatActivity, SettingsActivity::class.java).also {
```

Comment:
the best approach to this is to make a static method in the SettingsActivity called getIntent or something like that. By doing so each activity knows how to create itself and from here you will call only activity?.startActivity(SettingsActivity.getNewIntent(this))

Reply:
Got it

Reply:
Done